emote/memory/memory.py

```diff
@@ -169,3 +171,24 @@ def __iter__(self):
 
             data[self.size_key] = self.rollout_count * self.rollout_length
             yield {self.data_group: data, self.size_key: data[self.size_key]}
+
+
+class MemoryWarmup(Callback):
+    """A blocker to ensure memory has data.
+    
+    This ensures the memory has enough data when training starts, as the memory 
+    will panic otherwise. This is useful if you use an async data generator. 
+    
+    If you do not use an async data generator this will deadlock your training 
+    loop and prevent progress. 
+    """
+    
+    def __init__(self, loader: MemoryLoader):
+        super().__init__()
```

Comment:
Maybe set self._order to 100 or something. Then this will run after the GymCollector and we won't deadlock in practice.